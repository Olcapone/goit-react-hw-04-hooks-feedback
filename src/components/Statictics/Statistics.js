import React from 'react'
import { number, shape } from 'prop-types'
import shortid from 'shortid'
import s from './Statistics.module.css'
import Notification from '../Notification/Notification'

const Statistics = ({ options, total, positivePercentage }) => {
  const newArr = Object.keys(options)
  return (
    <>
      {total >= 1
        ? (
        <ul className={s.list}>
          {newArr.map((option) => (
            <li className={s.items} key={shortid.generate()}>
              {option}:<span className={s.goodStats}>{options[option]}</span>
            </li>
          ))}
          <li className={s.items}>
            Total:
            <span className={s.goodStats}>{total}</span>
          </li>
          <li className={s.items}>
            Positive feedback:
            <span
              className={positivePercentage > 50 ? s.goodStats : s.badStats}
            >
              {positivePercentage} %
            </span>
          </li>
        </ul>
          )
        : (
        <Notification message="No feedback given"></Notification>
          )}
    </>
  )
}

Statistics.propTypes = {
  total: number,
  positivePercentage: number,
  //  options: array
  options: shape({})
}

export default Statistics
