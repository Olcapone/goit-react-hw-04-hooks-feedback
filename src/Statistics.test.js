
import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';

import Statistics from '../src/components/Statictics/Statistics';
import userEvent from '@testing-library/user-event';

const props =  [{id:"1", name: "Jane"}, {id:"2", name: "Ann"}];
//jest.mock('../src/components/Statictics/Statistics', () => Promise.resolve(props));

describe("render Statistics", () => {

  it("should expose loading and posts prop", () => {
    const optionsCallbackMock = jest.fn();
  //   const { getByTestId } = render(
  //     <Statistics>{optionsCallbackMock}</Statistics>
  //   );
  //  // await screen.findByText(/No feedback given/i);
  //   expect(screen.queryByText(/Total/i)).toBeNull();
  //   expect(optionsCallbackMock).toHaveBeenCalledWith({
  //     loading: false,
  //     options: props
  //   })
    //userEvent.type(screen.getByRole('list'), 'Total');
    //fireEvent.change(screen.getByRole("list"), {target: {value: "good"}});
    //expect(screen.queryByText(/Total/i)).toBeInTheDocument();

    
    
    // const {queryByText, getByText} = render(<Statistics options={[...props]} />); 
    // const listElement = queryByText(/Total/i);
    // expect(listElement).toBeNull();
    
    
    // const { asFragment } = render(<Statistics options="test options" />); 
    // expect(asFragment(<Statistics/>)).toMatchSnapshot();
    // render(<Statistics options="no options" />);
  
  })

 it('has no props', () => {
    const {getByText} = render(<Statistics options="no options" />); 
    const listElement = getByText(/No feedback given/i);
    expect(listElement).toBeInTheDocument();

  })
  
});


