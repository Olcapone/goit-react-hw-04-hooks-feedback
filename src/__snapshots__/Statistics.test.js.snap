// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics has props 1`] = `
"<Fragment>
  <Notification message=\\"No feedback given\\" />
</Fragment>"
`;

exports[`render Statistics has props 1`] = `
<DocumentFragment>
  <h2
    class="title"
  >
    No feedback given
  </h2>
</DocumentFragment>
`;
